client/src/Features/MicrosoftLearn/MicrosoftLearnFilterUtils.ts

```diff
@@ -0,0 +1,145 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Product, Role, Level, Catalog } from '../../Models/Learn';
+import { FilterType } from '../../Models/Learn/FilterType.model';
+import { LearnTypeFilterOption, FilterOption } from './MicrosoftLearnFilterComponentProps';
+import _ from 'lodash';
+import { getParentProduct, scoreRegex } from './MicrosoftLearnFilterCore';
+import { Filter } from '../../Models/Learn/Filter.model';
+
+const FilterOptionComparer = (a: FilterOption, b: FilterOption) => {
+  if (a && b) {
+    return a.name.localeCompare(b.name);
+  } else if (a) {
+    return 1;
+  }
+  return -1;
+};
+
+const createOptoinArrayFromKeys = (keys: FilterOption[]) => {
```

Comment:
please see if this helps
```suggestion
const createOptionArrayFromKeys = (keys: FilterOption[]) => {
```